libafl_targets/src/sancov_cmp.rs

```diff
@@ -23,3 +29,67 @@ extern "C" {
     pub fn __sanitizer_cov_trace_switch(val: u64, cases: *const u64);
 
 }
+
+#[no_mangle]
+unsafe extern "C" fn __sanitizer_cov_pcs_init(pcs_beg: *const usize, pcs_end: *const usize) {
+    // "The Unsafe Code Guidelines also notably defines that usize and isize are respectively compatible with uintptr_t and intptr_t defined in C."
+    assert!(
+        PCS_BEG.is_null(),
+        "__sanitizer_cov_pcs_init can be called only once."
+    );
+    assert!(
+        PCS_END.is_null(),
+        "__sanitizer_cov_pcs_init can be called only once."
+    );
+
+    PCS_BEG = pcs_beg;
+    PCS_END = pcs_end;
+}
+
+/// An entry to the sanitizer_cov pc_table
+#[repr(C, packed)]
+#[derive(Debug, PartialEq, Eq)]
+pub struct PcTableEntry {
+    addr: usize,
+    flags: usize,
+}
+
+impl PcTableEntry {
+    /// Returns whether the PC corresponds to a function entry point.
+    pub fn is_function_entry(&self) -> bool {
+        self.flags == 0x1
+    }
+
+    /// Returns the address associated with this PC.
+    pub fn addr(&self) -> usize {
+        self.addr
+    }
+}
+
+/// Returns a slice containing the PC table.
+pub fn sanitizer_cov_pc_table() -> Option<&'static [PcTableEntry]> {
+    unsafe {
+        if PCS_BEG.is_null() || PCS_END.is_null() {
+            return None;
+        }
+        let len = PCS_END.offset_from(PCS_BEG);
+        assert!(
+            len > 0,
```

Comment:
if `len` is 0, should we return `None` or abort?

Reply:
Up to you. To me, a len of 0 here would mean a big error somewhere. It would mean that there's not even a point in collecting coverage (the coverage map size would be 0 too).

Note that it's not even an error of the kind of "I forgot to insturment the binary", because in that case PCS_BEG and PCS_END would be null (`__sanitizer_cov_pcs_init` would not be called if you specify `pc-table` but not any kind of coverage).